eng/common/pipelines/templates/steps/docsms-ensure-validation.yml

```diff
@@ -0,0 +1,21 @@
+steps:
+# Fail the build if any of the packages failed validation. Valid values are
+# "true" or "false" though some attempt is made to parse the boolean value from
+# the string.
+- pwsh: |
+    $value = '$(DocsMsPackagesAllValid)'
+
+    try {
+        $result = [System.Convert]::ToBoolean($value)
+        if (!$result) {
+            Write-Error "Some packages failed validation"
+            exit 1
+        }
+
+        Write-Host "All packages passed validation"
+    } catch [FormatException] {
+        Write-Host "Failed to parse DocsMsPackagesAllValid value '$value' as a boolean."
+        Write-Error $_
+        exit 1
+    }
```

Comment:
Does base exception fall through?

Reply:
When you do `Write-Error "Some packages failed validation"` this throws an exception which the `catch` block then catches. A better way to do this would be `Write-Host` and then `exit 1`

Reply:
I meant if it didnt match the FormatException type. Looks like we'll still throw.